File: sources/iOS/ios-communications/Sources/PolarBleSdk/sdk/api/model/PolarDeviceLog.swift
// Copyright © 2026 Polar Electro Oy. All rights reserved.

import Foundation

/// Represents a log file fetched from a Polar device.
///
/// - `path`: the file path on the device (e.g. "/ERRORLOG.BPB")
/// - `data`: the raw file contents
public struct PolarDeviceLog {
    /// The file path on the device (e.g. "/ERRORLOG.BPB")
    public let path: String
    /// The raw file contents
    public let data: Data

    public init(path: String, data: Data) {
        self.path = path
        self.data = data
    }
}

File: sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/api/model/PolarDeviceLog.kt
// Copyright © 2026 Polar Electro Oy. All rights reserved.
package com.polar.sdk.api.model

/**
 * Represents a log file fetched from a Polar device.
 *
 * @property path the file path on the device (e.g. "/ERRORLOG.BPB")
 * @property data the raw file contents as a [ByteArray]
 */
data class PolarDeviceLog(
    val path: String,
    val data: ByteArray
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is PolarDeviceLog) return false
        return path == other.path && data.contentEquals(other.data)
    }

    override fun hashCode(): Int {
        var result = path.hashCode()
        result = 31 * result + data.contentHashCode()
        return result
    }
}
